RelayTestApp/Assets/Scripts/brainCloud/BrainCloudManager.cs

```diff
@@ -0,0 +1,443 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using BrainCloud.JsonFx.Json;
+using UnityEngine;
+using BrainCloud;
+using BrainCloud.UnityWebSocketsForWebGL.WebSocketSharp;
+
+public class BrainCloudManager : MonoBehaviour
+{
+    private BrainCloudWrapper m_bcWrapper;
+    private bool m_dead = false;
+    public bool LeavingGame;
+    public BrainCloudWrapper Wrapper => m_bcWrapper;
+    public static BrainCloudManager Instance;
+    //Offset for the different mouse coordinates from Unity space to Nodejs space
+    private float _mouseYOffset = 321;
+    private void Awake()
+    {
+        m_bcWrapper = GetComponent<BrainCloudWrapper>();
+        if (!Instance)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    //Called from Unity Button, attempting to login
+    public void Login()
+    {
+        string username = GameManager.Instance.UsernameInputField.text;
+        string password = GameManager.Instance.PasswordInputField.text;
+        if (username.IsNullOrEmpty())
+        {   
+            StateManager.Instance.AbortToSignIn($"Please provide a username");
+            return;
+        }
+        if (password.IsNullOrEmpty())
+        {
+            StateManager.Instance.AbortToSignIn($"Please provide a password");
+            return;
+        }
+        
+        GameManager.Instance.CurrentUserInfo.Username = username;
+        InitializeBC();
+        // Authenticate with brainCloud
+        m_bcWrapper.AuthenticateUniversal(username, password, true, HandlePlayerState, LoggingInError, "Login Failed");
+    }
+
+    private void FixedUpdate()
+    {
+        if (m_bcWrapper != null && !LeavingGame)
+        {
+            m_bcWrapper.Update();
+        }
+        
+
+        if (m_dead)
+        {
+            m_dead = false;
+            
+            UninitializeBC();
+        }
+    }
+
+    private void InitializeBC()
+    {
+        string url = BrainCloud.Plugin.Interface.DispatcherURL;
+        string appId = BrainCloud.Plugin.Interface.AppId;
+        string appSecret = BrainCloud.Plugin.Interface.AppSecret;
+        m_bcWrapper.Init(url, appSecret, appId, "1.0");
```

Comment:
I believe if you just call "Init()" with no argument, it will use the plugin values by default so you dont have to get them like that.

Reply:
_you can verify that this is true_